src/py21cmfast/wrapper.py

```diff
@@ -2540,6 +2553,32 @@ def run_lightcone(
                 """
             )
 
+        coeval_callback_output = []
+        if coeval_callback is not None:
+            if isinstance(coeval_callback_redshifts, (list, np.ndarray)):
+                compute_coeval_callback = [False for i in range(len(scrollz))]
+                for coeval_z in coeval_callback_redshifts:
+                    assert isinstance(coeval_z, (int, float, np.number))
+                    compute_coeval_callback[
+                        np.argmin(np.abs(scrollz - coeval_z))
+                    ] = True
+                if sum(compute_coeval_callback) != len(coeval_callback_redshifts):
+                    logger.warning(
+                        "some of the coeval_callback_redshifts refer to the same node_redshift"
+                    )
+            elif (
+                isinstance(coeval_callback_redshifts, int)
+                and coeval_callback_redshifts > 0
+            ):
+                compute_coeval_callback = [
+                    True if i % coeval_callback_redshifts == 0 else False
```

Comment:
this would be simpler as `[not i % coeval_callback_redshifts for i in range(len(scrollz))]`